homeassistant/components/qbittorrent/sensor.py

```diff
@@ -4,122 +4,171 @@
+            coordinator, ["downloading", "uploading"]
+        ),
+    ),
+    QBittorrentSensorEntityDescription(
+        key=SENSOR_TYPE_INACTIVE_TORRENTS,
+        translation_key="inactive_torrents",
+        native_unit_of_measurement="torrents",
+        value_fn=lambda coordinator: count_torrents_in_states(
+            coordinator, ["stalledDL", "stalledUP"]
+        ),
+    ),
+    QBittorrentSensorEntityDescription(
+        key=SENSOR_TYPE_PAUSED_TORRENTS,
+        translation_key="paused_torrents",
+        native_unit_of_measurement="torrents",
+        value_fn=lambda coordinator: count_torrents_in_states(
+            coordinator, ["pausedDL", "pausedUP"]
+        ),
     ),
 )
 
 
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
-    async_add_entites: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up qBittorrent sensor entries."""
+
     coordinator: QBittorrentDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
-    entities = [
-        QBittorrentSensor(description, coordinator, config_entry)
+
+    async_add_entities(
+        QBittorrentSensor(coordinator, config_entry, description)
         for description in SENSOR_TYPES
-    ]
-    async_add_entites(entities)
+    )
 
 
 class QBittorrentSensor(CoordinatorEntity[QBittorrentDataCoordinator], SensorEntity):
     """Representation of a qBittorrent sensor."""
 
+    _attr_has_entity_name = True
     entity_description: QBittorrentSensorEntityDescription
 
     def __init__(
         self,
-        description: QBittorrentSensorEntityDescription,
         coordinator: QBittorrentDataCoordinator,
         config_entry: ConfigEntry,
+        entity_description: QBittorrentSensorEntityDescription,
     ) -> None:
         """Initialize the qBittorrent sensor."""
         super().__init__(coordinator)
-        self.entity_description = description
-        self._attr_unique_id = f"{config_entry.entry_id}-{description.key}"
-        self._attr_name = f"{config_entry.title} {description.name}"
-        self._attr_available = False
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{config_entry.entry_id}-{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, config_entry.entry_id)},
+            manufacturer="QBittorrent",
+        )
 
     @property
     def native_value(self) -> StateType:
-        """Return value of sensor."""
-        return self.entity_description.value_fn(self.coordinator.data)
+        """Return the value of the sensor."""
+        return self.entity_description.value_fn(self.coordinator)
+
+
+def count_torrents_in_states(
+    coordinator: QBittorrentDataCoordinator, states: list[str]
+) -> int:
+    """Count the number of torrents in specified states."""
+    return len(
```

Comment:
This is a calculation of state and not really allowed in integrations that integrate devices or services. These integrations should represent state as is returned from the API in question while following our entity design guidelines.